experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyTree.ts

```diff
@@ -408,20 +410,41 @@ export class LazyMap<TSchema extends MapSchema>
 		) as TypedField<TSchema["mapFields"]>;
 	}
 
-	// TODO: when appropriate add setter that delegates to field kind specific setter.
-	// public set(key: FieldKey, content: FlexibleFieldContent<TSchema["mapFields"]>): void {
-	// 	const field = this.get(key);
-	// 	if (field.is(SchemaBuilder.optional(this.schema.mapFields.allowedTypes))) {
-	// 		field.setContent(content);
-	// 	} else {
-	// 		assert(
-	// 			field.is(SchemaBuilder.sequence(this.schema.mapFields.allowedTypes)),
-	// 			"unexpected map field kind",
-	// 		);
-	// 		// TODO: fix merge semantics.
-	// 		field.replaceRange(0, field.length, content as Iterable<ContextuallyTypedNodeData>);
-	// 	}
-	// }
+	public set(
+		key: FieldKey,
+		content: FlexibleFieldContent<TSchema["mapFields"]> | undefined,
+	): void {
+		const field = this.getBoxed(key);
+		if (
+			field.is(
```

Comment:
The type guard is syntactically nice here, but having to create a `TreeFieldSchema` instance just to do the check is a little strange, not to mention the comparison logic of `schema.equals` inside of `field.is`. We already know that the allowed types match. The only thing we're not sure about is the field kind. So another option would be to simply switch on the `kind` of our schema, and then cast to the kind of field you know it is.

Reply:
Nice! Good call!